tests/epyccel/recognised_functions/test_numpy_types.py

```diff
@@ -292,15 +295,21 @@ def get_int8(a):
     b = int8(a)
     return b
 
-@pytest.mark.parametrize( 'function_boundaries', [(get_int, min_int, max_int), (get_int64, min_int64, max_int64), (get_int32, min_int32, max_int32),\
-                                                 (get_int16, min_int16, max_int16), (get_int8, min_int8, max_int8)])
+if numpy_basic_types_deprecated:
+    int_functions_and_boundaries = [(get_int64, min_int64, max_int64), (get_int32, min_int32, max_int32),\
+                                                 (get_int16, min_int16, max_int16), (get_int8, min_int8, max_int8)]
+else:
+    int_functions_and_boundaries = [(get_int, min_int, max_int), (get_int64, min_int64, max_int64), (get_int32, min_int32, max_int32),\
+                                                 (get_int16, min_int16, max_int16), (get_int8, min_int8, max_int8)]
+
+@pytest.mark.parametrize( 'function_boundaries', int_functions_and_boundaries)
 def test_numpy_int_scalar(language, function_boundaries):
 
     integer8 = randint(min_int8, max_int8, dtype=np.int8)
-    integer16 = randint(min_int16, max_int16, dtype=np.int16)
-    integer = randint(min_int, max_int, dtype=int)
-    integer32 = randint(min_int32, max_int32, dtype=np.int32)
-    integer64 = randint(min_int64, max_int64, dtype=np.int64)
+    integer16 = randint(min_int8, max_int8, dtype=np.int16)
+    integer = randint(min_int8, max_int8, dtype=int)
+    integer32 = randint(min_int8, max_int8, dtype=np.int32)
+    integer64 = randint(min_int8, max_int8, dtype=np.int64)
```

Comment:
shouldn't these use min and max for the associated types?

Reply:
No. The test converts integers of precision X to int8 objects. If the original int is larger than the target int then there will be overflow warnings. The latest version of numpy flags these overflow warnings with a `DeprecationError`, indicating that in a few versions time, this code will fail if we pass an int that is out of the bounds `[min_int8, max_int8]` to the function `int8`

Reply:
ah ok i see :+1: